integrations/tensorflow/bindings/python/pyiree/tf/support/tf_test_utils.py

```diff
@@ -565,68 +571,40 @@ def compile_module(
     exported_names: optional iterable of strings representing which of
       module_class's functions to compile. If exported_names is empty all
       functions will be compiled.
-
-  Returns:
-    Class decorator function.
   """
 
-  def decorator(cls):
-    """Decorator Function."""
-    if not issubclass(cls, TracedModuleTestCase):
-      logging.exception(
-          "The 'compile_module' decorator must be applied to a "
-          "TracedModuleTestCase derived class, which %s is not.", cls)
-    cls._module_class = module_class
-    cls._exported_names = exported_names
-    return cls
-
-  return decorator
-
-
-# Will be initialized by TracedModuleTestCase.setUpClass
-# Global variables are used because storing the compiler context on the cls
-# causes cleaning up refcounts to fail, and tf.test.TestCase wipes the variables
-# on the class instance (self.*) before each unittest.
-# TODO(#2900): Move these back to class variables when we figure out issues with
-# refcounting.
-_global_ref_module = None
-_global_tar_modules = None
+  # Setup the directory for saving compilation artifacts and traces.
+  global _global_artifacts_dir
+  _global_artifacts_dir = _setup_artifacts_dir(module_class.__name__)
+
+  # Get the backend information for this test.
+  ref_backend_info = tf_utils.BackendInfo(FLAGS.reference_backend,
+                                          f"{FLAGS.reference_backend}_ref")
+  tar_backend_infos = get_target_backends()
+
+  compile_backend = lambda backend_info: backend_info.compile(
+      module_class, exported_names, _global_artifacts_dir)
```

Comment:
Context: The matching function call in the to-be-added `tf_test_utils.compile_signature_def_saved_model` is really long, and it was more readable to write it as a lambda than to try to put it into a list-comprehension or for-loop. I wrote this this way to be consistent between the two functions.